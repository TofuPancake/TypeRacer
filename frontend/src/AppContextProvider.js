import React, { useState, useEffect } from 'react';
import { auth } from './firebase';
import themeMusic from './sound/theme_music.mp3';

const AppContext = React.createContext();

function AppContextProvider({ children }) {
    const [firebaseUserIdToken, setFirebaseUserIdToken] = useState(null);
    const [username, setUsername] = useState('');
    const [lobbyid, setLobbyid] = useState('');
    const [lobby, setLobby] = useState(null);
    const audio = new Audio(themeMusic);
    audio.loop = true;
    audio.volume = 0.1;

    useEffect(() => {
        auth.onIdTokenChanged(async (auth) => {
            if (auth) {
                const token = await auth.getIdToken();
                setFirebaseUserIdToken(token);
            } else {
                setFirebaseUserIdToken(null);
            }
        });
    }, []);

    const context = {
        firebaseUserIdToken,
        username,
        setUsername,
        lobbyid,
        setLobbyid,
        lobby,
        setLobby,
        audio
    }

    return (
        <AppContext.Provider value={context}>
            {children}
        </AppContext.Provider>
    );
}

export {
    AppContext,
    AppContextProvider,
}
