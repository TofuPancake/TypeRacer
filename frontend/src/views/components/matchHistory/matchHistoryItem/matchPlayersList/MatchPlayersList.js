import { useState } from 'react';
import { Avatar, List, makeStyles, ListItem, ListItemAvatar, ListItemText } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
    list: {
        width: '18vw',
        marginLeft: '5px',
        justifySelf: 'inline-end',
        minWidth: '100px',
    },
    listItem: {
        paddingTop: '0px',
        paddingBottom: '0px',
    },
    small: {
        width: theme.spacing(3),
        height: theme.spacing(3),
        marginRight: '0px',
    },
    listItemText: {
        alignSelf: 'center',
        justifySelf: 'flex-end',
        width: '30%',
        marginLeft: '3%',
        marginTop: '1px',
        marginBottom: '1px',
    }
}));

export default function MatchPlayersList({players}){
    const classes = useStyles();
    const [selectPlayer, setSelectPlayer] = useState();

    return (
        <List className={classes.list}>
        {
            players.map((player) => {
                let style = {};
                if (selectPlayer === player._id) {
                style['backgroundColor'] = 'rgb(230, 230, 230)';
                style['borderRadius'] = '12px';
                }
                return(
                    <div key={player._id}>
                        <ListItem className={classes.listItem} alignItems='center' style={style} onMouseEnter={() => setSelectPlayer(player._id)}>
                            <ListItemAvatar>
                                {player && player.photoURL ? <Avatar src={player.photoURL} className={classes.small} /> : <Avatar className={classes.small} />}
                            </ListItemAvatar>
                            <ListItemText className={classes.listItemText} primary={`${player.name}`} secondary={` WPM: ${player.wpm}`} />
                        </ListItem>
                    </div>
                );
            })
        }
        </List>
    )
}