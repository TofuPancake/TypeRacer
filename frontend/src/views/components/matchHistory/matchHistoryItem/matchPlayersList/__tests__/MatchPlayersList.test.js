import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { AppContextProvider } from '../../../../../../AppContextProvider';

import MatchPlayersList from '../MatchPlayersList';

test('match players list renders correctly', () => {
  const shallowRenderer = new ShallowRenderer();
  const snapshotComponent = shallowRenderer.render(
    <AppContextProvider>
      <MatchPlayersList />
    </AppContextProvider>);
  expect(snapshotComponent).toMatchSnapshot();
});