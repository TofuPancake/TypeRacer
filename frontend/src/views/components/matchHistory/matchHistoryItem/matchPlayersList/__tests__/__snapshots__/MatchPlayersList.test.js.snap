// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match players list renders correctly 1`] = `
<Context.Provider
  value={
    Object {
      "audio": <audio
        loop=""
        preload="auto"
        src="mock file"
      />,
      "firebaseUserIdToken": null,
      "lobby": null,
      "lobbyid": "",
      "setLobby": [Function],
      "setLobbyid": [Function],
      "setUsername": [Function],
      "username": "",
    }
  }
>
  <MatchPlayersList />
</Context.Provider>
`;
